src/components/restore/RestoreIcloudStep.js

```diff
@@ -0,0 +1,300 @@
+import { useNavigation } from '@react-navigation/native';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import {
+  Alert,
+  Dimensions,
+  InteractionManager,
+  KeyboardAvoidingView,
+  Platform,
+  StatusBar,
+  View,
+} from 'react-native';
+import ShadowStack from 'react-native-shadow-stack/dist/ShadowStack';
+import { useDispatch } from 'react-redux';
+import styled from 'styled-components';
+import { saveUserBackupState } from '../../handlers/localstorage/globalSettings';
+import { removeWalletData } from '../../handlers/localstorage/removeWallet';
+import BackupStateTypes from '../../helpers/backupStateTypes';
+import isNativeStackAvailable from '../../helpers/isNativeStackAvailable';
+import WalletLoadingStates from '../../helpers/walletLoadingStates';
+import {
+  useAccountSettings,
+  useInitializeWallet,
+  useWallets,
+} from '../../hooks';
+import { fetchBackupPassword, saveBackupPassword } from '../../model/keychain';
+import { restoreCloudBackup } from '../../model/wallet';
+import { sheetVerticalOffset } from '../../navigation/effects';
+import Routes from '../../navigation/routesNames';
+import { usePortal } from '../../react-native-cool-modals/Portal';
+import { setIsWalletLoading, walletsLoadState } from '../../redux/wallets';
+import { borders, colors, padding } from '../../styles';
+import { deviceUtils } from '../../utils';
+import { RainbowButton } from '../buttons';
+import { Icon } from '../icons';
+import { Input } from '../inputs';
+import { Column, Row } from '../layout';
+import LoadingOverlay, { LoadingOverlayWrapper } from '../modal/LoadingOverlay';
+import { GradientText, Text } from '../text';
+
+const sheetHeight = deviceUtils.dimensions.height - 108;
+
+const SheetContainer = isNativeStackAvailable
```

Comment:
We're about to unify it no?

Reply:
This PR is so big already that I'd prefer to handle that separately and get this merged